hikari/impl/interaction_server.py

```diff
@@ -0,0 +1,597 @@
+        await runner.shutdown()
+        await runner.cleanup()
+        future.set_result(None)
+
+    async def join(self) -> None:
+        if not self._runner:
+            raise TypeError("Cannot wait for an inactive interaction server to join")
+
+        await self._future
+
+    async def on_interaction(self, body: bytes, signature: bytes, timestamp: bytes) -> interaction_server.Response:
+        if not self._application_id or not self._verify:
+            self._application_id, self._verify = await self._fetch_application_information()
+
+        if not self._verify(body, signature, timestamp):
+            _LOGGER.info("Received a request with an invalid signature")
+            return _Response(_BAD_REQUEST_STATUS, "Invalid request signature")
+
+        if self._check_duplication(body):
+            _LOGGER.warning("Ignoring duplicated interaction request\n  %r", body)
+            return _Response(_OK_STATUS, _ACK_PAYLOAD, content_type=_JSON_CONTENT_TYPE)
+
+        try:
+            payload = self._loads(body.decode("utf-8"))
+            interaction_type = int(payload["type"])
+
+        except (data_binding.JSONDecodeError, ValueError) as exc:
+            _LOGGER.info("Received a request with an invalid JSON body", exc_info=exc)
+            return _Response(_BAD_REQUEST_STATUS, "Invalid JSON body")
+
+        except (KeyError, TypeError) as exc:
+            _LOGGER.info("Invalid or missing 'type' field in received JSON payload", exc_info=exc)
+            return _Response(_BAD_REQUEST_STATUS, "Invalid or missing 'type' field in payload")
+
+        if interaction_type == _PING_INTERACTION_TYPE:
+            _LOGGER.debug("Responding to ping interaction")
+            return _Response(_OK_STATUS, _PONG_PAYLOAD, content_type=_JSON_CONTENT_TYPE)
+
+        try:
+            event = self._event_factory.deserialize_interaction_create_event(
+                None, payload, application_id=self._application_id
+            )
+
+        except Exception as exc:
+            asyncio.get_running_loop().call_exception_handler(
+                {"message": "Exception occurred during interaction deserialization", "exception": exc}
+            )
+            return _Response(_INTERNAL_SERVER_ERROR_STATUS, "Exception occurred during interaction deserialization")
+
+        _LOGGER.debug("Dispatching interaction %s", event.interaction.id)
+
+        if self._events:
+            asyncio.create_task(
+                self._events.dispatch(event), name=f"{event.interaction.id} interaction request dispatch"
+            )
+
+        result: typing.Optional[special_endpoints.InteractionResponseBuilder] = None
+        if self._listener:
+            try:
+                result = await self._listener(event)
+
+            except Exception as exc:
+                asyncio.get_running_loop().call_exception_handler(
+                    {"message": "Exception occurred during interaction dispatch", "exception": exc}
+                )
```

Comment:
Cant we just log this normally? And if you do keep it like this, please use the loop first and then call `call_exception_handler` for redability